### DISCUSSION TITLE
Feature: Order the highlights based on the selected date

### DISCUSSION THREAD

**Discussion by adiati98** [open-sauced/feedback#22] [General feedback or Bugs]
### Type of feature

🍕 Feature

### Current behavior

Currently, when we create or edit a highlight and select a date, the latest created/edited highlight goes to the top (see attached gif).

![edit highlights open sauced](https://github.com/open-sauced/insights/assets/45172775/bd4dcb4e-3c1f-45ca-adaa-45c0abb73475)

### Suggested solution

It would be nice if the order of the highlights are based on the selected date from the latest to the oldest (or vice versa).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

**Comment by a0m0rajab** [comment]
Just to make sure, do you want to have a button to order/sort highlights on the user page?

**Comment by adiati98** [comment]
@a0m0rajab not necessary to have a button. But it would be great to have the highlights sorted based on the selected date to see the timeline of the PR clearly.

**Comment by bdougie** [comment]
We added the calendar based on user feedback and haven't enhanced that experience since we didn't get much feedback since. I would need to hear more context the use case add dates and would love to hear adding filter for select and created dates makes sense.

**Comment by isabensusan** [comment]
I think @adiati98 makes a good point.

### tl;dr
In **user profiles** we should sort Highlights in chronological order, using the dates selected by the user. By showing the highlights as a timeline, it better tells the story of this particular contributor and their OS journey. For the feed page though, I'd leave it as is (show and sort by highlight creation date).

### Long explanation
Highlights reference two different dates: 
1. Highlight date: the date in which the highlight was created in OpenSauced.
2. PR date: the date in which the PR referenced in the highlight was created, that the user can select in the calendar.

If I sign up to OpenSauced today and want to post highlights for my 3 most meaningful contributions (which could have happened months back) the Highlight date will be the same for all of them (today), and they will be sorted in the order in which I created the highlights.

But what happens if:
Example 1 - My most meaningful PR was a contribution to the React repo when it was just getting started. A contribution made to React in 2014 is going to weigh very differently vs a contribution made in 2023, and we should be showing that.

Example 2 - One of my first big contributions fixed a very important problem in a repo so I want to post a highlight about it, but the way I wrote code back then isn't as good as my current coding skills. I have another, more recent contribution with better code that I also want to highlight. I would want the highlights to show that some time has passed in between both contributions so that it is also clear that I have leveled-up my coding skills.

I can think of more use cases, but imo these are the two most relevant.

### Suggested Solution
Like @adiati98 mentioned:
1. On **user profiles** show highlights as a timeline, sort chronologically displaying the date of the PR (we can also show the date in which the highlight was created, just less prominently)
2. On the **highlights feed page** keep it as is showing how long it's been since the highlight was published, and add the PR date somewhere to the card.

**Comment by bdougie** [comment]
Thanks for these thoughts @isabensusan. Adding historical contributions that might fall outside our (undocumented) 90 days contribution window is a good enough reason to explore this. OpenSauced needs more signals like this to now how far back in time folks would like to go to build their resume of contributions. 

I think this issue from @CBID2 is slightly related, sort by date could tie well into grouping highlights later. 

For those doing the 100 days of open source may want create stories or collections of highlights. Just linking that for future reference.

**Comment by CBID2** [comment]
Thanks for tagging me @bdougie. For context @adiati98, this is the issue Brian is talking about: https://github.com/open-sauced/insights/issues/1392

**Comment by adiati98** [comment]
Thank you for explaining it so well, @isabensusan! Those are exactly what I meant 😃.

Yes, it's slightly related to the issue from @CBID2, @bdougie.
